assets/src/scripts/components/graph/view/water-levels.js

```diff
@@ -10,29 +11,33 @@ const CIRCLE_RADIUS_SINGLE_PT = 3;
  * @param  {Function} options.xScale    D3 scale function
  * @param  {Function} options.yScale    D3 scale function
  */
-export const drawDataLine = function (elem, {line, xScale, yScale}) {
+export const drawDataLine = function (elem, {line, xScale, yScale}, segment) {
     // If this is a single point line, then represent it as a circle.
     // Otherwise, render as a line.
     if (line.points.length === 1) {
-        elem.append('circle')
-            .data(line.points)
+        segment = segment || elem.append('circle')
             .classed('line-segment', true)
             .classed('approved', line.classes.approved)
             .classed('provisional', line.classes.provisional)
-            .attr('r', CIRCLE_RADIUS_SINGLE_PT)
+            .attr('r', CIRCLE_RADIUS_SINGLE_PT);
+        segment
+            .data(line.points)
+            .transition(transition().duration(100))
             .attr('cx', d => xScale(d.dateTime))
             .attr('cy', d => yScale(d.value));
     } else {
-        const tsLine = d3Line()
-            .x(d => xScale(d.dateTime))
-            .y(d => yScale(d.value));
-        elem.append('path')
-            .datum(line.points)
+        segment = segment || elem.append('path')
             .classed('line-segment', true)
             .classed('approved', line.classes.approved)
             .classed('provisional', line.classes.provisional)
-            .attr('d', tsLine);
+            .attr('vector-effect', 'non-scaling-stroke');
+        segment
+            .datum(line.points)
+            .transition(transition().duration(100))
+            .attr('d', d3Line().x(d => xScale(d.dateTime))
+                               .y(d => yScale(d.value)));
     }
+    return segment;
```

Comment:
Same question as before, why are you returning segment? And if there is a reason the doc comments should be updated.

Reply:
Yep, will update the jsdoc for places like this.